functional-tests/appwrapper_reconciler_test.go

```diff
@@ -0,0 +1,251 @@
+	// initiate setting up the EnvTest environment
+	testEnv = startEnvTest(t)
+
+	establishClient(t, testEnv) // setup client required for test to interact with kubernetes machine API
+
+	test := WithConfig(t, cfg)
+
+	// create namespace using envtest client in accordance with created kubeconfig's namespace name
+	client := test.Client()
+
+	//read machineset yaml from file `test_instascale_machineset.yml`
+	b, err := os.ReadFile("test_instascale_machineset.yml")
+	test.Expect(err).ToNot(HaveOccurred())
+
+	//deserialize kubernetes object
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	ms, _, err := decode(b, nil, nil) //decode machineset content of YAML file into kubernetes object
+	test.Expect(err).ToNot(HaveOccurred())
+	msa := ms.(*machinev1beta1.MachineSet) //asserts that decoded object is of type `*machinev1beta1.MachineSet`
+
+	//create machineset in default namespace
+	ms, err = client.Machine().MachineV1beta1().MachineSets("default").Create(test.Ctx(), msa, metav1.CreateOptions{})
+	test.Expect(err).ToNot(HaveOccurred())
+
+	//assert that the replicas in Machineset specification is 0, before appwrapper creation
+	machineset, err := client.Machine().MachineV1beta1().MachineSets("default").Get(test.Ctx(), "test-instascale", metav1.GetOptions{})
+	test.Expect(err).ToNot(HaveOccurred())
+	test.Expect(machineset.Spec.Replicas).To(gstruct.PointTo(Equal(int32(0))))
+
+	// create MCADclient for interacting with cluster API using cfg provided
+	mcadClient, err := mc.NewForConfig(cfg)
+	test.Expect(err).ToNot(HaveOccurred())
+
+	//create new test namespace
+	namespace := test.NewTestNamespace()
+
+	// initializes an appwrapper in the created namespace
+	aw := instascaleAppwrapper(namespace.Name)
+
+	// create appwrapper resource using mcadClient
+	aw, err = mcadClient.WorkloadV1beta1().AppWrappers(namespace.Name).Create(test.Ctx(), aw, metav1.CreateOptions{})
+	test.Expect(err).ToNot(HaveOccurred())
+	time.Sleep(10 * time.Second)
+
+	//update appwrapper status to avoid appwrapper dispatch in case of Insufficient resources
+	aw.Status = mcadv1beta1.AppWrapperStatus{
+		Pending: 1,
+		State:   mcadv1beta1.AppWrapperStateEnqueued,
+		Conditions: []mcadv1beta1.AppWrapperCondition{
+			{
+				Type:    mcadv1beta1.AppWrapperCondBackoff,
+				Status:  apiv1.ConditionTrue,
+				Reason:  "AppWrapperNotRunnable",
+				Message: "Insufficient resources to dispatch AppWrapper.",
+			},
+		},
+	}
+	_, err = mcadClient.WorkloadV1beta1().AppWrappers(namespace.Name).UpdateStatus(test.Ctx(), aw, metav1.UpdateOptions{})
+	test.Expect(err).ToNot(HaveOccurred())
+	time.Sleep(10 * time.Second)
+
+	// assert for machine replicas belonging to the machine set after appwrapper creation- there should be 1
+	machinesetafter, err := client.Machine().MachineV1beta1().MachineSets("default").Get(test.Ctx(), "test-instascale", metav1.GetOptions{})
+	test.Expect(err).ToNot(HaveOccurred())
+	test.Expect(machinesetafter.Spec.Replicas).To(gstruct.PointTo(Equal(int32(1))))
```

Comment:
Can you replace this part with `test.Eventually(......<implementation checking that within some timeout the replicas value is 1>)`?